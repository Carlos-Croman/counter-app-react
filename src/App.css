.app-title {
  margin-top: 0;
  margin-bottom: 3rem;
  font-size: 3.6rem;
  text-align: center;
}

.counter {
  margin-block: 2rem;
  font-size: 2.4rem;
  font-weight: 600;
  text-align: center;
}

.buttons-container {
  display: flex;
  justify-content: space-evenly;
}

.button {
  border-radius: 1rem;
  border: 1px solid transparent;
  padding: 0.8rem 1.6rem;
  font-size: 1.8rem;
  font-weight: 500;
  background-color: #1a1a1a;
  color: rgb(255, 255, 255, 75%);
  cursor: pointer;
  transition: border-color 0.25s, transform 0.25s;
}
.button:hover {
  border-color: #646cff;
}
.button:active {
  transform: scale(0.9);
}