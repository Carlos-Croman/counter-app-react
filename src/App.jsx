import { useState } from "react";
import PropTypes from "prop-types";
import "./App.css";

function App({ appTitle, counterValue }) {
  const [counter, setCounter] = useState(counterValue);

  return (
    <>
      <header>
        <h1 className="app-title">{ appTitle }</h1>
      </header>
      <div>
        <p className="counter">{ counter }</p>
      </div>
      <div className="buttons-container">
        <button className="button" onClick={() => setCounter(() => counter + 1)}>
          +1
        </button>
        <button className="button" onClick={() => setCounter( counterValue )}>
          &#x27F3;
        </button>
        <button className="button" onClick={() => setCounter(() => counter - 1)}>
          -1
        </button>
      </div>
    </>
  );
}

App.propTypes = {
  appTitle: PropTypes.string.isRequired,
  counterValue: PropTypes.number
}

App.defaultProps = {
  counterValue: 0,
}

export default App;