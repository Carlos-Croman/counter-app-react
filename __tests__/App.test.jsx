import { fireEvent, render, screen } from "@testing-library/react";
import { describe, test, expect } from "vitest";
import App from "../src/App";

const appTitle = "Counter App";
const counterValue = 10;

describe("<App /> - unit test", () => {
  test("must match snapshot", () => {
    const { container } = render(<App appTitle={appTitle} counterValue={counterValue} />);

    expect(container).toMatchSnapshot();
  });

  test("should increment the counter on click add button", () => {
    // Arrange
    render(<App appTitle={appTitle} />);

    // Act
    fireEvent.click(screen.getByRole("button", { name: "+1" }));

    // Assert
    expect(screen.getByText("1")).toBeInTheDocument();
  });

  test("should reset the counter to its initial value on click reset button", () => {
    render(<App appTitle={appTitle} counterValue={counterValue} />);

    fireEvent.click(screen.getByRole("button", { name: "+1" }));
    fireEvent.click(screen.getByRole("button", { name: "âŸ³" }));

    expect(screen.getByText(counterValue)).toBeInTheDocument();
  });

  test("should decrement the counter on click subtract button", () => {
    render(<App appTitle={appTitle} counterValue={counterValue} />);

    fireEvent.click(screen.getByRole("button", { name: "-1" }));
    fireEvent.click(screen.getByRole("button", { name: "-1" }));

    expect(screen.getByText(`${counterValue - 2}`)).toBeInTheDocument();
  });
});