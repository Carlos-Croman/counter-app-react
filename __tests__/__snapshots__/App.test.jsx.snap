// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App /> - unit test > must match snapshot 1`] = `
<div>
  <header>
    <h1
      class="app-title"
    >
      Counter App
    </h1>
  </header>
  <div>
    <p
      class="counter"
    >
      10
    </p>
  </div>
  <div
    class="buttons-container"
  >
    <button
      class="button"
    >
      +1
    </button>
    <button
      class="button"
    >
      ‚ü≥
    </button>
    <button
      class="button"
    >
      -1
    </button>
  </div>
</div>
`;
